python/pyarrow/parquet.py

```diff
@@ -1674,6 +1686,12 @@ def pieces(self):
     keys and only a hive-style directory structure is supported. When
     setting `use_legacy_dataset` to False, also within-file level filtering
     and different partitioning schemes are supported.
+pre_buffer : bool, default True
+    Coalesce and issue file reads in parallel to improve performance on
```

Comment:
Is there any impact/conflict with specifying `use_threads=False` ?

Reply:
There's no conflict: use_threads controls whether decoding work is done in parallel on the CPU thread pool, and pre_buffer controls whether I/O is done in parallel on the I/O thread pool. I guess for someone who wants Arrow to be truly single-threaded, this may be confusing, so I'll see if I can reword this.

Also, #9620 has some refactoring to allow pre-buffering without requiring use of the I/O thread pool.

Reply:
> I guess for someone who wants Arrow to be truly single-threaded, this may be confusing, so I'll see if I can reword this.

So for such a case, you would need to set `pre_buffer=False` to have it truly single-threaded?  
I am thinking about dask, where `use_threads` is set to False, but I am not sure if it would be a problem for them to still do I/O in parallel.

Reply:
Yes, you'd need to set both to False to get true single-threading. I added a short note about the use of threads to the docstring.